Association refset component validation

dionmcm opened the pull request:
According to https://confluence.ihtsdotools.org/display/DOCRELFMT/5.2.5.1+Historical+Association+Reference+Sets association reference sets such as REPLACED BY can refer to any component, not just concepts that this test is limited to.

This change expands this test to consider references to descriptions and relationships as valid too, this is necessary for the Australian Extension which does refer to inactive descriptions and their replacements which appears to be in specification for RF2.

dionmcm committed:
Updated to refer to components as allowed by the RF2 spec rather than just concepts

dionmcm committed:
fix tabs in file-centric-snapshot-association-valid-targetcomponentid.sql

CoderMChu commented:
@dionmcm Thanks for the PR. It works but I am just a bit worried about the performance as it is now linking lots of data.(Concepts, Descriptions and relationships)
It might perform better if we split this into two assertions. One for validating REPLACED BY only. The other one for remaining association types.(i.e concepts only) What do you think?

dionmcm commented:
Yes, that makes sense

dionmcm commented:
Actually, from this table in the spec it looks like the following is the actual valid types

Association reference set | Source type | Target type
-- | -- | --
900000000000523009 \| POSSIBLY EQUIVALENT TO association reference set\| | Concept | Concept
900000000000524003 \| MOVED TO association reference set\| | Component | Component
900000000000525002 \| MOVED FROM association reference set\| | Component | Component
900000000000526001 \| REPLACED BY association reference set\| | Component | Component
900000000000527005 \| SAME AS association reference set\| | Component | Component
900000000000528000 \| WAS A association reference set\| | Concept | Concept
900000000000529008 \| SIMILAR TO association reference set\| | (not used currently)
900000000000530003 \| ALTERNATIVE association reference set\| | Concept | Concept
900000000000531004 \| REFERS TO concept association reference set\| | Description | Concept

Do you want to split it into an assertion based on source/target type?

Type | Members
--|--
Concept/Concept | POSSIBLY EQUIVALENT TO, WAS A, ALTERNATIVE
Component/Component | MOVED TO, MOVED FROM, REPLACED BY, SAME AS
Description/Concept | REFERS TO

Looks like SAME AS, POSSIBLY EQUIVALENT TO, MOVED TO, WAS A, and REPLACED BY are the big ones.

dionmcm committed:
updated to split concept centric refsets from component centric refsets

dmcgihtsdo commented:
Hi @dionmcm, 

Many thanks for highlighting and fixing this issue. 

Part of the solution has now been covered by VAL-394 ( https://github.com/IHTSDO/snomed-release-validation-assertions/commit/05d0a72e87f9b768be6b59a50e797e9f527ed213 ). 

The relationship clause has been excluded, but a follow up ticket has been logged ( VAL-401 ).

Thanks.
